### PR TITLE
Unit tool tip fixes

### PR DESCRIPTION
**Pull request by kuronekochomusuke**
- fix issues with mismatched htlm tags in the unit and pilot tool tips.
- isolate the setting of fonts in in the unit and pilot tool tips, so that it would be easier to replace them with CSS.
- fix issue with hard coded font colors in unit and pilot tool tips.  use system defined ones instead.
- fix issue with the unicode characters in the movement section of unit tool tip.
- add in and `<hr>` around the pilot info in the board view tool tip, to make it it does not blend in with the unit info.
- fix issue when determining good critial slots, it did not include missing ones.  so it was diplaying double (good and bad), when missing.
- only show the hits chance section in the attack tool tip when in the firing or physical phase.  in the reporting phases only show who did it, since the hit info is no longer valid.

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/boardview/AttackSprite.java [hunk 1]
```diff
@@ -402,16 +406,19 @@ public void addWeapon(SearchlightAttackAction attack) {
...
-        tipString.append(Integer.toHexString(attackColor.getRGB() & 0xFFFFFF));
-        tipString.append(">");
-        tipString.append(attackerDesc
-                + "<BR>&nbsp;&nbsp;" + Messages.getString("BoardView1.on") + "&nbsp;&nbsp;" + targetDesc);
-        tipString.append("</FONT>");
-        for (String wpD: weaponDescs) {
-            tipString.append("<BR>"+wpD);
+        GamePhase phase = this.boardView1.game.getPhase();
+        String result = "";
+        String f = "";
```
Comment (rjhancock) [inline megamek/src/megamek/client/ui/swing/boardview/AttackSprite.java:411]: Please use more descriptive variable names. This provides no information on what this variable does.

File: megamek/src/megamek/client/ui/swing/boardview/BoardView.java [hunk 2]
```diff
@@ -5453,22 +5452,30 @@ public String getHexTooltip(MouseEvent e) {
...
+
         Hex mhex = game.getBoard().getHex(mcoords);
 
-        StringBuffer txt = new StringBuffer(HTML_BEGIN);
-        txt.append("<div WIDTH=500>");
+        String b = "";
+        String f = "";
+        String t = "";
+        String result = "";
+        String i = "";
```
Comment (rjhancock) [inline megamek/src/megamek/client/ui/swing/boardview/BoardView.java:5462]: Please use more descriptive variable names. This provides no information on what this variable does.

File: megamek/src/megamek/client/ui/swing/boardview/BoardView.java [hunk 3]
```diff
@@ -5710,23 +5759,26 @@ public void appendTerrainTooltip(StringBuffer txt, @Nullable Hex mhex) {
         }
 
         Coords mcoords = mhex.getCoords();
-        txt.append("<TABLE BORDER=0 BGCOLOR=" + TERRAIN_BGCOLOR + " width=100%><TR><TD><FONT color=\"black\">");
 
-        txt.append(Messages.getString("BoardView1.Tooltip.Hex", mcoords.getBoardNum(), mhex.getLevel()));
-        txt.append("<br>");
+        String result = "";
+        String f = "";
```
Comment (rjhancock) [inline megamek/src/megamek/client/ui/swing/boardview/BoardView.java:5764]: Please use more descriptive variable names. This provides no information on what this variable does.

File: megamek/src/megamek/client/ui/swing/boardview/BoardView.java [hunk 4]
```diff
@@ -5737,103 +5789,118 @@ public void appendBuildingsTooltip(StringBuffer txt, @Nullable Hex mhex) {
             return;
         }
         Coords mcoords = mhex.getCoords();
+        String f = "";
+        String result = "";
```
Comment (rjhancock) [inline megamek/src/megamek/client/ui/swing/boardview/BoardView.java:5793]: Please use more descriptive variable names. This provides no information on what this variable does.

**Comment by rjhancock** [comment]
Overall doesn't look bad but please go through your changes and these files and adjust the variable names to be more descriptive.

**Comment by HammerGS** [comment]
Want to confirm all fixes are in place, then I'll merge.

**Review by rjhancock — APPROVED** [review]

**Pull request merged**